src/pfe/portal/modules/Templates.js

```diff
@@ -109,67 +127,152 @@ module.exports = class Templates {
     return this.projectTemplates;
   }
 
-  async getTemplatesOfStyle(projectStyle) {
-    const allTemplates = await this.getTemplateList();
-    const relevantTemplates = allTemplates.filter(template =>
-      getTemplateStyle(template) === projectStyle
-    );
-    return relevantTemplates;
-  }
-
-  writeRepositoryList() {
-    // Use a callback here so we don't block the response to the request.
-    fs.writeJson(this.repositoryFile, this.repositoryList, { spaces: '  ' }, err => {
-      if (err) {
-        log.error(`Error writing repository list to ${this.repositoryFile}: ${err}`)
-      }
+  // Save the default list to disk so the user can potentially edit it (WHEN CODEWIND IS NOT RUNNING)
+  async writeRepositoryList() {
+    try {
+      await fs.writeJson(this.repositoryFile, this.repositoryList, { spaces: '  ' });
       log.info(`Repository list updated.`);
-    });
+    } catch (err) {
+      log.error(`Error writing repository list to ${this.repositoryFile}: ${err}`)
+    }
   }
 
   getRepositories() {
     return this.repositoryList;
   }
 
+  getEnabledRepositories() {
+    return this.getRepositories().filter(repo =>
+      // if the repo doesn't specify whether it's enabled, consider it enabled
+      (repo.enabled || !repo.hasOwnProperty('enabled'))
+    );
+  }
+
+  doesRepositoryExist(repoUrl) {
+    try {
+      this.getRepository(repoUrl);
+      return true;
+    } catch (error) {
+      return false;
+    }
+  }
+
+  getRepositoryIndex(url) {
+    const repos = this.getRepositories();
+    const index = repos.findIndex(repo => repo.url === url);
+    return index;
+  }
+
+  /**
+   * @param {String} url
+   * @return {JSON} reference to the repo object in this.repositoryList
+   */
+  getRepository(url) {
+    const index = this.getRepositoryIndex(url);
+    if (index < 0) throw new Error(`no repository found with URL '${url}'`);
```

Comment:
Why throw error here rather than return `null` simply?  Seems like we're using error to control the flow when it's just as easy to do a `if` check.

Reply:
(As below) I changed `getRepository` to throw an error because I thought that being unable to get a repo is exceptional behaviour. I thought it best practice to throw an error rather than returning false, so that we get explicit information about why the call failed.

Reply:
Well, there's a performance cost to throwing errors.  Personally I tend to throw errors only in circumstance where something really exceptional happens and it's hard to detect/prevent ahead of time, and/or error throwing is an expected behaviour of some framework.

This doesn't seem to meet that bar since there are lots of cases where you try to get something from a collection (e.g. getting something from a map) and it doesn't exist.  In most of those case we don't throw errors.

Reply:
Ok, if I changed this back to returning a boolean instead of throwing an error, then for example, at https://github.com/eclipse/codewind/pull/122/files#diff-2e95eceb73a0346dfd44f09698ce42b3R45, rather than 
```
try {
    await templates.addRepository(extension.templates, extension.description);
} catch (error) {
   log.warn(error);
   // ignore so that we can try to add other repos in the list
}
```
we have to do
```
const success = await templates.addRepository(extension.templates, extension.description);
if (success === false) {
   log.warn('some message');
   // ignore so that we can try to add other repos in the list
}
```
which I think odd tbh (for a function that doesn't return data to return true/false). I also think that we won't notice the perf difference here, as the user calls this very rarely.

So until we notice the performance cost, I prefer the clearer code and more direct error message.

Ofc, if a load of people outweigh my opinion/ other reasons come up then I'm happy to change this

Reply:
I think the above comment was mean for `addRepository` function below? 

In any case, to summarize my thoughts on this matter:
- I still feel it's odd is for a collection getter to throw an error if something doesn't exist (could be wrong but this doesn't seem like a common practice)
- a method returning a boolean to indicate if the method was successful isn't that uncommon
- performance cost may not be noticeable here but every small thing adds up so it's always good keep it in mind

In the grand scheme of things, whether we throw error or return boolean is just a difference in philosophy and as long as the logic of the code we're doing here is correct, I'm not going block the change if you decide to keep things as-is.